src/view/mod.rs

```diff
@@ -10,39 +14,220 @@ use tui::{
     Terminal,
 };
 
+#[derive(Debug)]
+pub struct StyleFunc {
+    pub f: fn(Style) -> Style,
+}
+
 pub struct View<B: Backend> {
     terminal: Terminal<B>,
     // Probably want to include layout details
     //  here eventually.
+    tag_to_func: HashMap<Tag, StyleFunc>,
 }
 
 impl<B: Backend> View<B> {
-    pub fn new(terminal: Terminal<B>) -> View<B> {
-        View { terminal: terminal }
+    pub fn new(terminal: Terminal<B>, tag_to_func: HashMap<Tag, StyleFunc>) -> View<B> {
+        View {
+            terminal: terminal,
+            tag_to_func: tag_to_func,
+        }
     }
 
     pub fn update_display(&mut self, app: &App) -> Result<(), io::Error> {
-        // Get a copy of the text to be rendered
-        // For now let's not do anything fancy formatting
-        let text = app.get_text_based_on_mode();
-
-        let text: Vec<_> = text.iter().map(|x| Text::raw(x)).collect();
-
-        let title = match app.app_mode() {
-            AppMode::Command(CommandMode::Write) => "Command mode: Saving into file",
-            AppMode::Command(CommandMode::Read) => "Command mode: Opening a file",
-            AppMode::Edit => "Edit mode",
-        };
-
+        let tagged_text = app.get_tagged_text(); // Get a copy of the text to be rendered
+                                                 // For now let's not do anything fancy formatting
+        let text: Vec<_> = highlighter::highlight_tagged_text(&tagged_text, &self.tag_to_func);
         self.terminal.draw(|mut f| {
             let size = f.size();
             let block = Paragraph::new(text.iter())
-                .block(Block::default().title(title).borders(Borders::ALL))
```

Comment:
Due to these changes, I can't seem to use the Open and Save file features anymore. I think it should be a quick change for the `get_text_based_on_mode()` function to call `as_tagged_text()` on the correct buffer.